### ISSUE TITLE
confusing about the index convention of W

### ISSUE THREAD

**Issue by anthony123** [makeyourownneuralnetwork/makeyourownneuralnetwork#9]
suppose a network of 3 input nodes and 4 hidden layer nodes(first hidden layer), so the dimension of W is 4x3 .  and W(4,1) is the link between the first input node to the fourth hidden layer.  But in the book , W(4x3) refers to  the link between the fourth input node to the first hidden layer. Is there something wrong OR am i mistaken?

**Comment by makeyourownneuralnetwork** [comment]
Hi - I also struggled with conventions for matrices and links. In fact I've struggled with it for 20+ years over computing .. where I've seen all kinds of conventions used in maths and computing texts.

The key thing is to remain consistent in your code or project or paper or blog.

Here I've tried to stick to a convention that says W_from,to so w_1,2 is the weight for the link from the first to the second node. The picture attached here shows examples.

Early editions of the book did have typos in the diagrams so maybe that is the cause of confusion? Get back to me if this doesn't help.

![backprop_error_example](https://cloud.githubusercontent.com/assets/17411198/25074429/4086db96-22f2-11e7-9388-94e4fd5b81cd.png)
